mucommander-registry/src/main/java/com/mucommander/commons/file/protocol/registry/SkopeoCommandExecutor.java

```diff
@@ -0,0 +1,156 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.commons.file.protocol.registry;
+
+import com.mucommander.commons.file.Credentials;
+import com.mucommander.commons.file.FileFactory;
+import com.mucommander.process.AbstractProcess;
+import com.mucommander.process.ProcessListener;
+import com.mucommander.process.ProcessRunner;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SkopeoCommandExecutor {
+	private static final Logger LOGGER = LoggerFactory.getLogger(SkopeoCommandExecutor.class);
+	
+    public final static String UTF8 = "UTF-8";
+
+    /**
+     * Check whether skopeo tool exists in the system.
+     */
+    public static boolean checkSkopeo() {
+        StringBuilder output = new StringBuilder();
+        try {
+            SkopeoCommandExecutor.execute(output, "", null);
+        } catch (IOException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * The inspect command fetches the repository's manifest and it is able to
+     * show you a docker inspect-like json output about a whole repository or a tag.
+     */
+    public static JSONObject inspect(String imageUrl, Credentials creds) throws IOException {
+        StringBuilder output = new StringBuilder();
+        SkopeoCommandExecutor.execute(output, "inspect", creds, imageUrl);
+        JSONParser parser = new JSONParser();
+        Object obj = null;
+        try {
+            obj = parser.parse(output.toString());
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return (JSONObject)obj;
+    }
+
+    /**
+     * Copy container images between various storage mechanisms (docker/oci/dir/etc).
+     */
+    public static String copy(String imageUrl, String imageDigest, Credentials creds) throws IOException {
+        String tempFolder = FileFactory.getTemporaryFolder() + imageDigest;
+        StringBuilder output = new StringBuilder();
+        SkopeoCommandExecutor.execute(output, "copy", creds, imageUrl, "dir:" + tempFolder);
+        return tempFolder;
+    }
+
+    public static void execute(StringBuilder outputBuffer, String command, Credentials creds, String... args) throws IOException {
+        LOGGER.debug("Executing skopeo command: " + command);
+        List<String> tokens = new ArrayList<>(Arrays.asList(
+                "skopeo",
+                "--insecure-policy",
+                "--override-os",
+                "linux",
```

Comment:
should this be set to the running operating system?

Reply:
It's for supporting downloading linux images under other OSes. In the future, we probably should add a select-box for choosing an OS.